go/private/actions/archive.bzl

```diff
@@ -56,10 +56,14 @@ def emit_archive(go, source = None, _recompile_suffix = ""):
         pre_ext = ".external"
     if _recompile_suffix:
         pre_ext += _recompile_suffix
-    out_lib = go.declare_file(go, ext = pre_ext + ".a")
+
+    name = ""
```

Comment:
I think this can be simplified to `name = source.library.name`: no if statement needed. If `name` is falsy, then `_child_name` will use `go._ctx.label.name`, as it does today.

Reply:
done.